lib/Service/Note.php

```diff
@@ -105,6 +105,9 @@ public function getData(array $exclude = []) : array {
 		if (!in_array('readonly', $exclude)) {
 			$data['readonly'] = $this->getReadOnly();
 		}
+		$data['internalPath'] = $this->file->getInternalPath();
+		$data['shareTypes'] = $this->noteUtil->getShareTypes($this->file);
+		$data['isShared'] = (bool) count($data['shareTypes']);
```

Comment:
Should allow to achieve the same without triggering heavy loading of all the share entries from the db:

```suggestion
		$data['shareTypes'] = $this->file->isShared();
```

Reply:
I have tried `$this->file->isShared()` but it always returns false.

Reply:
Right, it would only return true for incoming shares (so for the receiving user). Let's keep that then as it is as the files app seems to do it similarly.